app.js

```diff
@@ -0,0 +1,68 @@
+const express = require("express");
+const cors = require("cors");
+// const axios = require("axios");
+const body  = require("body-parser");
+const mySchool = require('./School')
+
+const port = 3000;
+
+const app = express();
+app.use(cors());
+app.use(body.urlencoded({extended: false}));
+
+// let mySchool = new School();
+
+const getAllClasses = (req, res) => {
+    console.log("these are the classes" + mySchool.classes)
+    res.json({
+        allClasses: mySchool.classes
+        
+    })
+}
+
+const checkIfClassExists = (req, res, next) =>  {
+        
+    if (req.class) {
```

Comment:
You are not checking here if the class exists, you are checking if there is a request at all. 
You need to key into the mySchool.classes object and check if the class already exists there, using `req.body.class` as a key you are looking for. 
It's a good move though - to use middleware :)